pyTigerGraph/gds/featurizer.py

```diff
@@ -0,0 +1,189 @@
+        if not tasks:
+            return "Attribute already exists"
+        # Create schema change job 
+        job_name = "add_{}_attr_{}".format(schema_type,random_string(6)) 
+        job = "USE GRAPH {}\n".format(self.conn.graphname) + "CREATE GLOBAL SCHEMA_CHANGE JOB {} {{\n".format(
+            job_name) + ''.join(tasks) + "}}\nRUN GLOBAL SCHEMA_CHANGE JOB {}".format(job_name)
+        # Submit the job
+        resp = self.conn.gsql(job)
+        status = resp.splitlines()[-1]
+        if "Failed" in status:
+            raise ConnectionError(status)
+        else:
+            print(status)
+        return 'Global schema change succeeded.'
+
+    
+    def _get_Params(self,name_of_query:str):
+        '''
+        return default query parameter by parsing the query header
+        '''
+        self.local_gsql_path = pjoin(os.path.dirname(
+            os.path.abspath(__file__)), "gsql", "featurizer", name_of_query + ".gsql")
+        with open(self.local_gsql_path) as infile:
+            _dict = {}
+            query = infile.read()
+        try:
+            input_params = query[query.find('(')+1:query.find(')')]
+            list_params =input_params.split(',')
+            for i in range(len(list_params)):
+                if "=" in list_params[i]:
+                    params_type = list_params[i].split('=')[0].split()[0]
+                    if params_type.lower() == 'float' or params_type.lower() == 'double':
+                        _dict[list_params[i].split('=')[0].split()[1]] = float(list_params[i].split('=')[1])
+                    if params_type.lower() == 'bool':
+                        _dict[list_params[i].split('=')[0].split()[1]] = bool(list_params[i].split('=')[1])
+                    if params_type.lower() == 'int':
+                        _dict[list_params[i].split('=')[0].split()[1]] = int(list_params[i].split('=')[1])
+                else:
+                    _dict[list_params[i].split()[1]] =  None
+        except:
+            print("The algorithm does not have any input parameter.")
+        self.params_dict[self.name_of_query] = _dict
+        return _dict  
+    
+    def runAlgorithm(self,name_of_query:str,params:dict = None, timeout:int=2147480) -> None:
+        # Initialize the params (if it is None) by parsing the query.
+        if params == None:
+            params = self._get_Params(name_of_query)
+            if 'v_type' in params.values():
+                params['v_type'] = self.conn.getVertexTypes()[0]
+            if 'e_type' in params.values():
+                params['e_type'] = self.conn.getEdgeTypes()[0]
+            print("Default parameters are:",params)
+        if None in params.values():
+            raise ValueError("Query parameters which are None need to be initialized.")
+        result = self.conn.runInstalledQuery(name_of_query, params,timeout=timeout)
+        if result != None:
+            return "Success!"
```

Comment:
As @parkererickson mentioned, might want to return the result.

Reply:
I changed accordingly